### PR TITLE
 fix (#minor); Morpho; Address QA issues 

### PR DESCRIPTION
**Pull request by harsh9200**
**Deployments:**
- https://okgraph.xyz/?q=harsh9200%2Fmorpho-aave-v2-ethereum
- https://okgraph.xyz/?q=harsh9200%2Fmorpho-compound-ethereum

### CODE REVIEW

File: subgraphs/morpho/src/helpers.ts [hunk 1]
```diff
@@ -75,99 +81,159 @@ export function createInterestRate(
...
+    financialMetrics.dailyLiquidateUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyWithdrawUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyRepayUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailySupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyTotalRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyTransferUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyFlashloanUSD = BIGDECIMAL_ZERO;
+
+    financialMetrics.totalValueLockedUSD = BIGDECIMAL_ZERO;
```
Comment (dmelotik) [inline subgraphs/morpho/src/helpers.ts:186]: I'm not a huge fan of the pattern of initializing everything to 0. It makes it easier to forget to update the fields after the snapshot is created. I would prefer we take the `protocol` entity and set any fields that we can to the `protocol` fields.

This was actually causing issues in some YAs

File: subgraphs/morpho/src/helpers.ts [hunk 2]
```diff
@@ -950,16 +1004,61 @@ function snapshotPosition(position: Position, event: ethereum.Event): void {
 export function updateProtocolPosition(
   protocol: LendingProtocol,
   market: Market,
-  morphoPositions: MorphoPositions
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (dmelotik) [inline subgraphs/morpho/src/helpers.ts:1007]: Why do we want to keep this parameter if it isn't being used?

File: subgraphs/morpho/src/utils/maths/AaveMath.ts
Comment (dmelotik) [inline subgraphs/morpho/src/utils/maths/AaveMath.ts:1]: Can you use `camelCase.ts` to name these files?

**Review by dmelotik — COMMENTED** [review]
Looks good! Few comments. Also, can you rebase, we had to make a change to the deployment.json structure for morpho to support our BE

**Comment by dmelotik** [comment]
Let's fix these issues and merge then tackle: https://github.com/messari/subgraphs/issues/2267

File: subgraphs/morpho/src/helpers.ts (hunk 2 above)
Comment (harsh9200) [reply subgraphs/morpho/src/helpers.ts:1007]: My bad, I will remove it

File: subgraphs/morpho/src/helpers.ts (hunk 1 above)
Comment (harsh9200) [reply subgraphs/morpho/src/helpers.ts:186]: I see. I will update this.

File: subgraphs/morpho/schema.graphql [hunk 3]
```diff
@@ -35,7 +35,7 @@ enum ProtocolType {
   # Will add more
 }
 
-type Token @entity @regularPolling {
+type Token @entity {
```
Comment (dmelotik) [inline subgraphs/morpho/schema.graphql:38]: We gotta add the directives back in

File: subgraphs/morpho/src/helpers.ts [hunk 4]
```diff
@@ -1080,3 +1187,40 @@ export function updateP2PRates(market: Market): void {
 
 export const getEventId = (hash: Bytes, logIndex: BigInt): Bytes =>
   hash.concat(Bytes.fromI32(logIndex.toI32()));
+
+export function updateProtocolTotalValueLockedUSD(
```
Comment (dmelotik) [inline subgraphs/morpho/src/helpers.ts:1191]: Should we also extend this function to also update `totalDepositBalanceUSD` and `totalBorrowBalanceUSD`?

File: subgraphs/morpho/src/mapping/morpho-compound/comptroller.ts [hunk 5]
```diff
@@ -24,17 +24,14 @@ export function handleNewBorrowCap(event: NewBorrowCap): void {
...
     .minus(BIGDECIMAL_ONE);
-  const marketList = _MarketList.load(MORPHO_COMPOUND_ADDRESS);
-  if (!marketList) {
-    log.error("[handleNewCloseFactor] Market list not found", []);
-    return;
-  }
-  for (let i = 0; i < marketList.markets.length; i++) {
-    const market = getMarket(marketList.markets[i]);
+
+  for (let i = 0; i < protocol.markets.length; i++) {
```
Comment (dmelotik) [inline subgraphs/morpho/src/mapping/morpho-compound/comptroller.ts:33]: Does it work to use `protocol.markets[]`? I had changed this in the PR I made bc it wasn't building...

That is kinda strange, it looks like the gh action builds just fine..

**Review by dmelotik — COMMENTED** [review]
- @harsh9200 can you update versions
- @bye43 once synced let's QA [this](https://github.com/messari/subgraphs/pull/2199#issuecomment-1624170512)

Almost done! THanks

File: subgraphs/morpho/src/helpers.ts (hunk 4 above)
Comment (harsh9200) [reply subgraphs/morpho/src/helpers.ts:1191]: Yeah, makes sense

File: subgraphs/morpho/src/mapping/morpho-compound/comptroller.ts (hunk 5 above)
Comment (harsh9200) [reply subgraphs/morpho/src/mapping/morpho-compound/comptroller.ts:33]: yeah, It is working fine for me.

**Comment by harsh9200** [comment]
The TVL issue is now fixed. Now, the `protocol.tvl` on 5/8/23 matches the sum of `marketDailySnaphshot.tvl`. [qurery](https://api.thegraph.com/subgraphs/name/messari/morpho-aave-v2-ethereum/graphql?query=%7B%0A++markets%28first%3A+10%2C+orderBy%3A+totalValueLockedUSD%2C+orderDirection%3Adesc%29+%7B%0A++++id%0A++++name%0A++++totalDepositBalanceUSD%0A++++totalBorrowBalanceUSD%0A++++totalValueLockedUSD%0A++%7D%0A%7D)

**Comment by dmelotik** [comment]
@harsh9200 looking at the deployment you linked (messari/morpho). The `protocol.TVL` is still ~690M, whereas we are expecting ~780M based on the marketSnapshots (or most recent snapshots) on 5/8/23... Maybe there is something wrong in our expected calculation, but I don't see a hole in our logic (as described here: https://github.com/messari/subgraphs/issues/2267#issue-1792057908)

**Comment by bye43** [comment]
@dmelotik I believe that snapshot balance is correct now. From harsh and I's discussion on discord:

> the snapshot was being saved before we updated the market entity the input token balance and the TVL was of previous trigger of the market. So in reality there was  a spike of TVL in stETH pool but at the end of that day the spike was back to normal.

So the initial 790mi was from a large deposit that ended up being withdrawn before the end of the day. Believe that what we are seeing now is correct.

File: subgraphs/morpho/schema.graphql [hunk 6]
```diff
@@ -72,7 +72,7 @@ enum RewardTokenType {
   STAKE
 }
 
-type RewardToken @entity(immutable: true) @regularPolling {
+type RewardToken @entity(immutable: true) {
```
Comment (dmelotik) [inline subgraphs/morpho/schema.graphql:75]: Can you make sure all of the directives stay in the code?

**Review by dmelotik — APPROVED** [review]
Approving and ready to merge once the schema comment is fixed. Thanks for the hard work @harsh9200

**Pull request merged**